code/go/0chain.net/blobbercore/reference/ref_walker.go

```diff
@@ -0,0 +1,146 @@
+package reference
+
+import (
+	"path"
+	"strings"
+)
+
+func NewRefWalkerFromPath(p string) *RefWalker {
+	return NewRefWalker(strings.Split(path.Clean(p), "/"))
+}
+
+// NewRefWalker wrap dirs as RefWaller
+func NewRefWalker(dirs []string) *RefWalker {
+
+	return &RefWalker{
+		items:  dirs,
+		length: len(dirs),
+		index:  0,
+	}
+}
+
+type RefWalker struct {
+	items  []string
+	length int
+	index  int
+}
+
+// Name get current dir name
+func (d *RefWalker) Name() string {
+	if d == nil {
+		return "/"
+	}
+
+	// current is root
+	if d.index == 0 {
+		return "/"
+	}
+
+	if d.index < d.length {
+		return d.items[d.index]
+	}
+
+	return "/"
+}
+
+// Path get current dir path
+func (d *RefWalker) Path() string {
+	if d == nil {
```

Comment:
Perhaps we should return error when d is nil, or when index is >= length, otherwise users will have to check the code to know what happened when the returned value is "".

Reply:
fixed

Reply:
fixed